addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -363,6 +363,13 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
             this.selectActionEl.append(option);
         });
 
+            // Remove the 'id' from form fields to ensure unique IDs are generated
+            // when multiple form snippets are added to the page.
+            for (const field of FormEditorRegistry.get(this.activeForm.website_form_key)
```

Comment:
This is obviously not a better place to code this, IMO.
I see, in [this line](https://github.com/odoo/odoo/pull/183883/files#diff-8b11d359e1213470895c7a9d28b45424520045089330be5713e9f3865ad16406R158) we are generating the uniqueId if id attribute is not exists. (why not every-time?) I think it should solve the problem but need check if there is any limitation to do that.

Reply:
@divy-odoo Indeed we can just remove the function that generates the unique id from the IF condition. It makes absolutely no difference to the current ID generation + it would solve the issue of duplicate ids in multiple form snippets without the need of this extra code.

The if condition was added as field.id is deleted in [onClone function](https://github.com/odoo/odoo/pull/183883/files#diff-8b11d359e1213470895c7a9d28b45424520045089330be5713e9f3865ad16406L958-L960) but it makes absolutely no difference to how many times the generateId function is called if its wrapped in that IF or not wrapped.

So I too think we should remove the generateId function from [this IF condition](https://github.com/odoo/odoo/pull/183883/files#diff-8b11d359e1213470895c7a9d28b45424520045089330be5713e9f3865ad16406R158) and generate the ID every time.